src/test/moves/upper_hand.test.ts

```diff
@@ -0,0 +1,100 @@
+import { allMoves } from "#app/data/move";
+import { MoveResult } from "#app/field/pokemon";
+import { Abilities } from "#enums/abilities";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import GameManager from "#test/utils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it, vi } from "vitest";
+
+describe("Moves - Upper Hand", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .moveset(Moves.UPPER_HAND)
+      .ability(Abilities.BALL_FETCH)
+      .battleType("single")
+      .disableCrits()
+      .enemySpecies(Species.MAGIKARP)
+      .enemyAbility(Abilities.BALL_FETCH)
+      .enemyMoveset(Moves.QUICK_ATTACK)
+      .startingLevel(100)
+      .enemyLevel(100);
+  });
+
+  it("should flinch the opponent before they use a priority attack", async () => {
+    await game.classicMode.startBattle([ Species.FEEBAS ]);
+
+    const feebas = game.scene.getPlayerPokemon()!;
+    const magikarp = game.scene.getEnemyPokemon()!;
+
+    game.move.select(Moves.UPPER_HAND);
+    await game.phaseInterceptor.to("BerryPhase");
+
+    expect(feebas.getLastXMoves()[0].result).toBe(MoveResult.SUCCESS);
+    expect(magikarp.isFullHp()).toBeFalsy();
+    expect(feebas.isFullHp()).toBeTruthy();
+  });
+
+  it.each([
+    { descriptor: "non-priority attack", move: Moves.TACKLE },
+    { descriptor: "status move", move: Moves.BABY_DOLL_EYES }
+  ])("should fail when the opponent selects a $descriptor", async ({ move }) => {
+    game.override.enemyMoveset(move);
+
+    await game.classicMode.startBattle([ Species.FEEBAS ]);
+
+    const feebas = game.scene.getPlayerPokemon()!;
+
+    game.move.select(Moves.UPPER_HAND);
+    await game.phaseInterceptor.to("BerryPhase");
+
+    expect(feebas.getLastXMoves()[0].result).toBe(MoveResult.FAIL);
+  });
+
+  it("should flinch the opponent before they use an attack boosted by Gale Wings", async () => {
+    game.override
+      .enemyAbility(Abilities.GALE_WINGS)
+      .enemyMoveset(Moves.GUST);
+
+    await game.classicMode.startBattle([ Species.FEEBAS ]);
+
+    const feebas = game.scene.getPlayerPokemon()!;
+    const magikarp = game.scene.getEnemyPokemon()!;
+
+    game.move.select(Moves.UPPER_HAND);
+    await game.phaseInterceptor.to("BerryPhase");
+
+    expect(feebas.getLastXMoves()[0].result).toBe(MoveResult.SUCCESS);
+    expect(magikarp.isFullHp()).toBeFalsy();
+    expect(feebas.isFullHp()).toBeTruthy();
+  });
+
+  it("should fail if the target has already moved", async () => {
+    game.override.enemyMoveset(Moves.TACKLE);
+    vi.spyOn(allMoves[Moves.TACKLE], "priority", "get").mockReturnValue(4);
```

Comment:
```suggestion
    game.override.enemyMoveset(Moves.ENDURE);
```
The protect moves are priority class 4, you can just use one of those.

Reply:
I wanted the move to be an attack so that the increased priority would be the only factor in making Upper Hand fail.